src/csharp/Microsoft.Spark/Sql/RowCollector.cs

```diff
@@ -30,14 +25,9 @@ public IEnumerable<Row> Collect(ISocketWrapper socket)
             Stream inputStream = socket.InputStream;
 
             int? length;
-            while (((length = SerDe.ReadBytesLength(inputStream)) != null) &&
-                (length.GetValueOrDefault() > 0))
+            while (((length = SerDe.ReadBytesLength(inputStream)) != null) && (length.GetValueOrDefault() > 0))
             {
-                MaxLengthReadStream readStream = s_slicedReadStream ??
-                    (s_slicedReadStream = new MaxLengthReadStream());
-
-                readStream.Reset(inputStream, length.GetValueOrDefault());
-                var unpickledObjects = PythonSerDe.GetUnpickledObjects(readStream);
+                object[] unpickledObjects = PythonSerDe.GetUnpickledObjects(inputStream, length.Value);
```

Comment:
Should `.Value` be `.GetValueOrDefault()` instead?  (as discussed above)

Reply:
#done